pkg/portal/ssh/ssh.go

```diff
@@ -159,29 +168,58 @@ func (s *SSH) New(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	port := ""
-	if s.env.IsLocalDevelopmentMode() {
-		port = "-p 2222 "
-	}
-
-	s.sendResponse(w, &response{
-		Command:  fmt.Sprintf("ssh %s%s@%s", port, username, host),
-		Password: password,
-	})
+	s.sendResponse(w, host, username, password, "", s.env.IsLocalDevelopmentMode())
 }
 
-func (s *SSH) sendResponse(w http.ResponseWriter, resp *response) {
-	b, err := json.MarshalIndent(resp, "", "    ")
+func (s *SSH) sendResponse(w http.ResponseWriter, hostname, username, password, error string, isLocalDevelopmentMode bool) {
+	w.Header().Set("Content-Type", "application/json")
+
+	enc := json.NewEncoder(w)
+	enc.SetEscapeHTML(false)
+	enc.SetIndent("", "    ")
+
+	if error != "" {
+		err := enc.Encode(response{Error: error})
+		if err != nil {
+			s.internalServerError(w, err)
+		}
+		return
+	}
+	fingerprint := cryptossh.FingerprintSHA256(s.hostPubKey)
+	command, err := createLoginCommand(isLocalDevelopmentMode, username, hostname, fingerprint)
+	resp := response{Command: command, Password: password}
+	if err != nil {
+		s.internalServerError(w, err)
+	}
+	err = enc.Encode(resp)
 	if err != nil {
 		s.internalServerError(w, err)
-		return
 	}
-
-	w.Header().Set("Content-Type", "application/json")
-	_, _ = w.Write(b)
 }
 
 func (s *SSH) internalServerError(w http.ResponseWriter, err error) {
 	s.log.Warn(err)
 	http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 }
+
+const (
+	sshCommand = "echo '{{ .Hostname }} ssh-rsa {{.PublicKeyFingerprint}}' > {{.Hostname}}_known_host ; " +
```

Comment:
Do we ever clean up these files or would they stay on the RP until we redeploy?

Reply:
this command is not run on the rp, nor the portal. This is executed on the saw

Reply:
Hey Jeremy, thanks for this. Can you elaborate on this comment? How would this change our current workflow of connecting?

Reply:
This is just changing the command we have to copy paste from the saw

Reply:
There is no change in the flow

Reply:
Ack, understand the flow now after review.